Xamarin.Forms.Platform.Android/Material/MaterialProgressBarRenderer.cs

```diff
@@ -1,83 +1,237 @@
+			if (_disposed)
 				return;
+			_disposed = true;
+
+			if (disposing)
+			{
+				_visualElementTracker?.Dispose();
+				_visualElementTracker = null;
+
+				_visualElementRenderer?.Dispose();
+				_visualElementRenderer = null;
+
+				if (Element != null)
+				{
+					Element.PropertyChanged -= OnElementPropertyChanged;
 
-			var color = Element.BackgroundColor;
-			if (color != Color.Default)
-				Control.ProgressBackgroundTintList = ColorStateList.ValueOf(color.ToAndroid());
+					if (Platform.GetRenderer(Element) == this)
+						Element.ClearValue(Platform.RendererProperty);
+				}
+			}
+
+			base.Dispose(disposing);
 		}
 
-		internal protected override void UpdateProgressColor()
+		protected virtual void OnElementChanged(ElementChangedEventArgs<ProgressBar> e)
 		{
-			if (Build.VERSION.SdkInt < BuildVersionCodes.Lollipop)
+			ElementChanged?.Invoke(this, new VisualElementChangedEventArgs(e.OldElement, e.NewElement));
+
+			if (e.OldElement != null)
 			{
-				base.UpdateProgressColor();
-				return;
+				e.OldElement.PropertyChanged -= OnElementPropertyChanged;
+			}
+
+			if (e.NewElement != null)
+			{
+				this.EnsureId();
+
+				if (_visualElementTracker == null)
+					_visualElementTracker = new VisualElementTracker(this);
+
+				e.NewElement.PropertyChanged += OnElementPropertyChanged;
+
+				UpdateProgress();
+				UpdateColors();
+
+				ElevationHelper.SetElevation(this, e.NewElement);
 			}
+		}
 
+		protected virtual void OnElementPropertyChanged(object sender, PropertyChangedEventArgs e)
+		{
+			ElementPropertyChanged?.Invoke(this, e);
+
+			if (e.PropertyName == ProgressBar.ProgressProperty.PropertyName)
+				UpdateProgress();
+			else if (e.PropertyName == ProgressBar.ProgressColorProperty.PropertyName || e.PropertyName == VisualElement.BackgroundColorProperty.PropertyName)
+				UpdateColors();
+		}
+
+		public override bool OnTouchEvent(MotionEvent e)
+		{
+			if (_visualElementRenderer.OnTouchEvent(e) || base.OnTouchEvent(e))
+				return true;
+
+			return _motionEventHelper.HandleMotionEvent(Parent, e);
+		}
+
+		void UpdateColors()
+		{
 			if (Element == null || Control == null)
 				return;
 
-			Color color = Element.ProgressColor;
-			if (color == Color.Default)
-				return;
+			Color progressColor = Element.ProgressColor;
+			Color backgroundColor = Element.BackgroundColor;
+
+			var defaultProgress = MaterialColors.Light.PrimaryColor;
+
+			if (progressColor.IsDefault && backgroundColor.IsDefault)
+			{
+				// reset everything to defaults
+				ProgressTintList = ColorStateList.ValueOf(defaultProgress);
+				ProgressBackgroundTintList = ColorStateList.ValueOf(defaultProgress);
+				ProgressBackgroundTintMode = PorterDuff.Mode.SrcIn;
+			}
+			else if (progressColor.IsDefault && !backgroundColor.IsDefault)
+			{
+				// handle the case where only the background is set
+				var background = backgroundColor.ToAndroid();
+
+				// TODO: Potentially override primary color to match material design.
```

Comment:
Are these TODOs something we need to address now?

Reply:
Same comment as below